gobblin-temporal/src/main/java/org/apache/gobblin/temporal/joblauncher/GobblinJobLauncher.java

```diff
@@ -150,17 +155,23 @@ protected void runWorkUnits(List<WorkUnit> workUnits) throws Exception {
       // Start the output TaskState collector service
       this.taskStateCollectorService.startAsync().awaitRunning();
 
+      Future<?> submitJobFuture = null;
```

Comment:
why do we need this?

Reply:
Becuase the initialization is within synchronized block and we are waiting on line 174, which is outside the synch block. This was the original bug that the wait was happening in synch block and cancel couldn't proceed because of the lock.